repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java

```diff
@@ -183,6 +185,10 @@ public void onChange(ChangedTypeDefs changedTypeDefs) throws AtlasBaseException
             if (CollectionUtils.isNotEmpty(changedTypeDefs.getCreatedTypeDefs())) {
                 for (AtlasBaseTypeDef typeDef : changedTypeDefs.getCreatedTypeDefs()) {
                     updateIndexForTypeDef(management, typeDef);
+                    if(!typeDef.getIndexCreated()) {
+                        attemptRollback(changedTypeDefs, management);
+                        throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Typedef {} not created", typeDef.getName());
```

Comment:
Why is this a `BAD_REQUEST` ? If failures are due to systemic issues, please consider changing the error code respectively